src/request.ts

```diff
@@ -33,27 +28,65 @@ export async function request<Query, Variables>(
   try {
     const headers = {
       Authorization: `Bearer ${ctx.apiKey}`,
+      'Content-Type': 'application/json',
     };
 
     const url = ctx.apiUrl || defaultUrl;
 
-    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
-    const { data: res, headers: responseHeaders } = await axios.post(
-      url,
-      {
+    const result = await ctx.fetch(url, {
+      method: 'POST',
+      headers,
+      body: JSON.stringify({
         query: query,
         variables: args.variables || null,
-      },
-      {
-        headers,
-      }
-    );
+      }),
+    });
+
+    const status = result.status;
+    const responseHeaders = result.headers;
+
+    let response;
```

Comment:
Does this get typed corrected?

Reply:
No - it's any but that's technically true (as anything could happen with the request) so I think that's ok?